packaging/generic-rpm/amazon-ecs-init.spec

```diff
@@ -73,25 +77,32 @@ install -m %{no_exec_perm} %{agent_image} %{buildroot}%{_cachedir}/ecs/
 mkdir -p %{buildroot}%{_sharedstatedir}/ecs/data
 
 install -m %{no_exec_perm} -D %{SOURCE2} $RPM_BUILD_ROOT/%{_unitdir}/ecs.service
+install -m %{no_exec_perm} -D %{SOURCE5} $RPM_BUILD_ROOT/%{_unitdir}/amazon-ecs-volume-plugin.service
+install -m %{no_exec_perm} -D %{SOURCE6} $RPM_BUILD_ROOT/%{_unitdir}/amazon-ecs-volume-plugin.socket
 
 %files
 %{_libexecdir}/amazon-ecs-init
 %{_mandir}/man1/amazon-ecs-init.1*
+%{_libexecdir}/amazon-ecs-volume-plugin
 %config(noreplace) %ghost %{_sysconfdir}/ecs/ecs.config
 %config(noreplace) %ghost %{_sysconfdir}/ecs/ecs.config.json
 %ghost %{_cachedir}/ecs/ecs-agent.tar
 %{_cachedir}/ecs/%{basename:%{agent_image}}
 %{_cachedir}/ecs/state
 %dir %{_sharedstatedir}/ecs/data
 %{_unitdir}/ecs.service
+%{_unitdir}/amazon-ecs-volume-plugin.service
+%{_unitdir}/amazon-ecs-volume-plugin.socket
 
 %post
 # Symlink the bundled ECS Agent at loadable path.
 ln -sf %{basename:%{agent_image}} %{_cachedir}/ecs/ecs-agent.tar
 %systemd_post ecs
+%systemd_post amazon-ecs-volume-plugin
 
 %postun
-%systemd_postun
+%systemd_postun ecs.service
+%systemd_postun amazon-ecs-volume-plugin
```

Comment:
Are we trying to restart the services when upgrade is done? Would `systemd_postun_with_restart` be more suitable? https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_systemd

Reply:
And a more general question - if we already have `post` which gets executed when new package installed, what's the benefit of having `postun`?

Reply:
Good catch, I take a look at the spec for al2 and we are doing `systemd_postun_with_restart` there.

My thought for question is `postun` get executed after `post` during update so adding it will be safe. I am not sure about the initial intention this is introduced though.